MathLib/LinAlg/PETSc/PETScVector.cpp

```diff
@@ -61,6 +64,10 @@ PETScVector::PETScVector(const PetscInt vec_size,
     }
 
     config();
+
+    for (PetscInt i=0; i<nghosts; i++)
+        _global_ids2local_ids_ghost.insert
+        (std::make_pair(ghost_ids[i], _size_loc + i));
```

Comment:
Took emplace to insert map term.
@TomFischer The files in directory LinAlg/PETSc were formatted by using astyle before. I will send another PR just about applying clang-format to the files in LinAlg/PETSc.